### DISCUSSION TITLE
Feedback: Migrations

### DISCUSSION THREAD

**Discussion by wcwcaseman** [mswjs/mswjs.io#421] [Q&A] [answered]
- [Open page](https://mswjs.io/docs/migrations)

## Feedback

We use internal types from version 1, MockedResponse, RestRequest, ResponseTransformer, ResponseComposition and there is no clear upgrade path for using these. Also can't get codemod to work at all. Having typescript examples would also be helpful. I think we have a complex setup but any help would be appreciated thanks.

**Comment by kettanaito** [comment, marked as answer]
Hi, @wcwcaseman. Thanks for the feedback!

I hope most of your questions will be answered in [Using with TypeScript](https://mswjs.io/docs/best-practices/typescript). This goes through custom resolvers, handlers, and other things that require a bit more granular typing. 

Specifically for your types:

- `MockedResponse` -> `Response` (global).
- `RestRequest` -> `HttpRequest`.
- `ResponseTransformer` -> gone, no concept of [transforming response anymore](https://mswjs.io/docs/migrations/1.x-to-2.x#response-declaration).
- `ResponseComposition` -> gone, see the previous point.

Let me know if you have any concrete examples you'd like help migrating to v2.
